geoapps/driver_base/driver.py

```diff
@@ -0,0 +1,57 @@
+#  Copyright (c) 2022 Mira Geoscience Ltd.
+#
+#  This file is part of geoapps.
+#
+#  geoapps is distributed under the terms and conditions of the MIT License
+#  (see LICENSE file at the root of this source code package).
+
+from __future__ import annotations
+
+import os
+from abc import ABC, abstractmethod
+
+from geoh5py.ui_json import InputFile
+from param_sweeps.generate import generate
+
+from geoapps.driver_base.params import BaseParams
+
+
+class BaseDriver(ABC):
+
+    _params_class = BaseParams
+    _validations = None
+
+    def __init__(self, params: BaseParams):
+        self.params = params
+
+    @abstractmethod
+    def run(self):
+        """Run the application."""
+        raise NotImplementedError
+
+    @classmethod
+    def start(cls, filepath):
+
```

Comment:
Should probably add docstrings and types.

Reply:
done